src/linalg/polar.rs

```diff
@@ -0,0 +1,158 @@
+#[cfg(feature = "serde-serialize")]
+use serde::{Deserialize, Serialize};
+
+use approx::AbsDiffEq;
+
+use crate::{DMatrix};
+use crate::linalg::SVD;
+use alga::general::{ComplexField};
+
+/// Polar Decomposition of a general matrix.
+#[cfg_attr(feature = "serde-serialize", derive(Serialize, Deserialize))]
+#[cfg_attr(
+    feature = "serde-serialize",
+    serde(bound(
+            serialize = "DMatrix<N>: Serialize,
+                        DMatrix<N>: Serialize,
+                        DMatrix<N>: Serialize"
+    ))
+)]
+#[cfg_attr(
+    feature = "serde-serialize",
+    serde(bound(
+            deserialize = "DMatrix<N>: Deserialize<'de>,
+                        DMatrix<N>: Deserialize<'de>,
+                        DMatrix<N>: Deserialize<'de>"
+    ))
+)]
+#[derive(Clone, Debug)]
+pub struct Polar<N: ComplexField>
+{
+    /// The rotation matrix
+    pub r: Option<DMatrix<N>>,
+    /// The left hermitian matrix (A = PR)
+    pub p_l: Option<DMatrix<N>>,
+    /// The right hermitian matrix (A = RP)
+    pub p_r: Option<DMatrix<N>>
+}
+
+impl<N: ComplexField> Polar<N>
+where
+{
+    /// Computes the Polar Decomposition of the matrix using its SVD
+    pub fn new(matrix: DMatrix<N>) -> Self {
+        Self::try_new(
+            matrix,
+            N::RealField::default_epsilon(),
+            0
+        ).unwrap()
+    }
+
+    /// Attempts to compute the Polar Decomposition using the SVD of the matrix
+    ///
+    /// # Arguments
+    ///
+    /// * `eps`           − tolerance used to determine when a value converged to 0.
+    /// * `max_niter`     − maximum total number of iterations performed by the algorithm. If this
+    /// number of iteration is exceeded, `None` is returned. If `niter == 0`, then the algorithm
+    /// continues indefinitely until convergence.
+    pub fn try_new(
+        matrix: DMatrix<N>,
+        eps: N::RealField,
+        max_niter: usize
+    ) -> Option<Self> {
+
+        let svd_opt = SVD::try_new(matrix, true, true, eps, max_niter);
+
+        if let Some(svd) = svd_opt {
```

Comment:
Here, if `svd_opt` is `None`, you want `try_new` to return `None` as well. So instead of a `if let { ... } else { None }` you can simply use the `?` operator:

```rust
let svd = SVD::try_new(matrix, true, true, eps, max_niter)?;
```

This will automatically return `None` if `SVD::try_new` returns `None`. You can learn more about the `?` operator [there](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html).